### PR TITLE
Hot fix#3566

### LINKED ISSUE #3566: Resource Calendar - Days and month are not translated
#### Issue

Days of week and months cannot be translated on the Resources Calendar

#### Expected Behavior

Days and month should be translated

#### Steps to Reproduce

1. Install a language pack
2. Login with that language
3. Enter Menu ALL / Resources Calendar
4. See days and month not translated

2017-05-17 12_15_04-projetos suitecrm

or a pic from another user and language
0921055e-2c03-11e7-9a86-17e603116628

Same goes for the resource calendar when you view a project in Gantt mode view (enter a Project / Actions tab/ View Gantt mode)
2017-05-24 12_06_48-suitecrm_gantt

#### Context

Also posted by @maladrill in a different but non related issue

#### Your Environment

* SuiteCRM Version used:
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): 7.8.x and 7.9.0 RC
* Environment name and version (e.g. MySQL, PHP 7): 
* Operating System and version (e.g Ubuntu 16.04):

### PR DESCRIPTION
**Pull request by darkassazi**
This pull request fixes #3566
Now  chart and gantt gonna use selected language

## Motivation and Context
When you don't know other language than native language :)

## How To Test This
go to project -> resource calendar
and go to project -> project list -> and select gantt chart.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [X] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [X] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

**Comment by pgorod** [comment]
You should be directing this PR at the `hotifx` branch, not the `master` branch.

You can still change that by editing your PR.

**Comment by darkassazi** [comment]
ok. I change it

**Comment by horus68** [comment]
@darkassazi there is a conflicting file:  modules/Project/chart.php
Please verify the original files from where you created your files.
Please note that hotfix branch files are more updated then master files, the branch from where your created. So you must reflect the hotfix branch file on your proposal, changing from there.

File: modules/Project/chart.php [hunk 1]
```diff
@@ -1,23 +1,48 @@
...
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must 
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM". 
  */
 
+if (!defined('sugarEntry') || !sugarEntry) {
+	die('Not A Valid Entry Point');
+}
+
+
 class chart {
```
Comment (daniel-samson) [inline modules/Project/chart.php]: Minor Formatting Issue: Please move the left curly brace '{' to the next line.

Example:
```
class chart
{
```

See https://suitecrm.com/wiki/index.php/Coding_Standards for details.

File: modules/Project/gantt.php [hunk 2]
```diff
@@ -1,24 +1,47 @@
...
+ * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not 
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must 
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM". 
  */
 
+if (!defined('sugarEntry') || !sugarEntry) {
+	die('Not A Valid Entry Point');
+}
+
 class Gantt {
```
Comment (daniel-samson) [inline modules/Project/gantt.php]: Minor Formatting Issue: Please move the left curly brace '{' to the next line.

Example:
```
class Gantt
{

```

File: modules/Project/gantt.php [hunk 3]
```diff
@@ -224,10 +249,10 @@ public function year_month($start_date, $end_date, $diff_interval=1)
 		$interval = new DateInterval('P1D'); // 1 month interval
         $period = new DatePeriod($begin, $interval, $end);
         $aResult = array();
-
+        
         foreach ( $period as $dt )
         {
-            $aResult[$dt->format('Y')][strftime("%B", $dt->getTimestamp())][$dt->format('j')] = strftime("%a", $dt->getTimestamp());
+        	$aResult[$dt->format('Y')][$GLOBALS['app_list_strings']['dom_cal_month_short'][$dt->format('n')]][$dt->format('j')] = $GLOBALS['app_list_strings']['dom_cal_day_short'][$dt->format('w')+1];
```
Comment (daniel-samson) [inline modules/Project/gantt.php]: Line is longer than 120 characters and it is also difficult to read.Consider assigning the $GLOBALS and $dt->format methods into variables and then use the variables instead.

```
public function <name>
{
    global $app_list_strings;
    // ... 
    foreach ($period as $dt) {
        // Perhaps give better names here
        $y = $dt->format('Y');
        $n = $app_list_strings[$dt->format('n')];
        $w = $dt->format('w') + 1;
        $j = $dt->format('j');
            
        $aResult[$y][$n][$j] = $app_list_strings['dom_cal_day_short'][$w];
    }
    // ...
}

```

File: modules/Project/chart.php [hunk 4]
```diff
@@ -316,7 +341,7 @@ public function draw($start_date, $end_date, $sel_projects ,$sel_users, $sel_con
 				foreach($months as $days){
 
 					foreach($days as $day => $d){
-						echo '<td class="inner_td"><div class="cell_width">'.$d[0].'</div></td>';//First letter of the days name shown
+						echo '<td class="inner_td"><div class="cell_width">'.$d.'</div></td>';//First letter of the days name shown
```
Comment (daniel-samson) [inline modules/Project/chart.php]: Minor Formatting Issue: Please move the left curly brace '{' to the next line.

File: modules/Project/chart.php [hunk 5]
```diff
@@ -628,47 +653,47 @@ public function year_week($start_date, $end_date)
 		foreach ( $period as $dt )
 		{
 			$count++;
-			$aResult[$dt->format('Y')][$dt->format('M')][] = $dt->format('W');
-		}
-
-		return array($aResult, $count);
-	}
+      $aResult[$dt->format('Y')][$GLOBALS['app_list_strings']['dom_cal_month_short'][$dt->format('n')]][] = $dt->format('W');
```
Comment (daniel-samson) [inline modules/Project/chart.php]: Line is longer than 120 characters and it is also difficult to read.Consider assigning the $GLOBALS and $dt->format methods into variables and then use the variables instead.

```
public function <name>
{
    global $app_list_strings;
    // ... 
    foreach ($period as $dt) {
        // Perhaps give better names here
        $n = $GLOBALS['app_list_strings']['dom_cal_month_short'][$dt->format('n')];
        $aResult[$dt->format('Y')][$n][] = $dt->format('W');
    }
    // ...
}

```

File: modules/Project/chart.php [hunk 6]
```diff
@@ -628,47 +653,47 @@ public function year_week($start_date, $end_date)
...
+    $begin = new DateTime( $start_date );
+    $end = new DateTime( $end_date);
+    $end->add(new DateInterval('P1D')); //Add 1 day to include the end date as a day
+    $interval = new DateInterval('P1D'); // 1 month interval
+    $period = new DatePeriod($begin, $interval, $end);
+    $aResult = array();
+    
+    foreach ( $period as $dt )
+    {        	
+      $aResult[$dt->format('Y')][mb_substr($GLOBALS['app_list_strings']['dom_cal_month_short'][$dt->format('n')],0, 3)][$dt->format('j')] = mb_substr($GLOBALS['app_list_strings']['dom_cal_day_short'][$dt->format('w')+1],0, 1);
```
Comment (daniel-samson) [inline modules/Project/chart.php]: Again over 120 lines see previous comment.

File: modules/Project/chart.php [hunk 7]
```diff
@@ -628,47 +653,47 @@ public function year_week($start_date, $end_date)
...
+    }
+  
+    return $aResult;
+  }
+
+  function get_weeks($start_date, $end_date)
+  {
+    $begin = new DateTime( $start_date );
```
Comment (daniel-samson) [inline modules/Project/chart.php]: Please remove spaces after ( and before )

example:
`DateTime($start_date);`

**Comment by darkassazi** [comment]
I tested it on version 7.9.0 and chart works, problem is with gantt or more exact with root\modules\Project\views\view.ganttchart.php. 
In version 7.9.0 @Haris Raheem commentted lines between 184 and 235, these lines, are responsible for generating gantt chart page. So becouse of this, full gant chart is not generating.

I tested my changes for gantt.php on SuiteCRM version 7.8.3 and they worked.

**Comment by Dillon-Brown** [comment]
Works as expected

**Comment by horus68** [comment]
Fiz works for Resources Calendar and Project Gantt mode calendar

**Comment by likhobory** [comment]
Project calendar in Gantt view mode - OK
Resources Calendar  -  there is a problem with Cyrillic symbols. 
When I testing with English or Portuguese langpacks there is no problems, but when I use Russian langpack or replace values in arrays 'dom_cal_day_short' and 'dom_cal_month_short' (in Portuguese langpack) with Russian short weekday/month values:

>         'dom_cal_day_short'=>array(
>                 '0'=>'',
>                 '1'=>'Вс',
>                 '2'=>'Пн',
>                 '3'=>'Вт',
>                 '4'=>'Ср',
>                 '5'=>'Чт',
>                 '6'=>'Пт',
>                 '7'=>'Сб',
>         ),

>     'dom_cal_month_short' => array(
>       '0' => '' ,
>       '1' => 'Январь' ,
>       '2' => 'Февраль' ,
>       '3' => 'Март' ,
>       '4' => 'Апрель' ,
>       '5' => 'Май' ,
>       '6' => 'Июнь' ,
>       '7' => 'Июль' ,
>       '8' => 'Август' ,
>       '9' => 'Сентябрь' ,
>       '10' => 'Октябрь' ,
>       '11' => 'Ноябрь' ,
>       '12' => 'Декабрь' 
>     ),

I see the same picture:

default

**Comment by pgorod** [comment]
@likhobory could this be a browser problem? Does it look the same in a different browser?

I tried in my browser, not with the characters coming from the language pack, but simply opening the browser's Inspector and pasting the russian characters directly into the place where the weekdays are (in an english install). It shows correctly. I wonder if in your system the character is correct when seen from within the Inspector?

russian

russian2

What I am looking to find out is if your character is already wrong in the HTML, before the browser displays it, or if it is just a browser display problem, or a font problem...

**Comment by horus68** [comment]
@likhobory no problem for (copy pasted your code into portuguese file) 
prtugueserussian

I also installed the language pack from Crowdin and it works fine, out of the box:
russianpack

**Comment by likhobory** [comment]
>  Does it look the same in a different browser?
Yes, tested on win7 with FF 52 and on win10 with FF 52 and MS Edge.

default


In other table rows Cyrillic characters are displayed correctly.

**Comment by likhobory** [comment]
> I also installed the language pack from Crowdin and it works fine
Hmm.. @horus68 I'll try it.

**Comment by likhobory** [comment]
Yes, I agree, after a clean installation on another server it works fine!

**Comment by samus-aran** [comment]
@darkassazi  - Just a note going forward make at least one of your commits the standard from the [How to Contribute](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM#Providing_a_Bug_Fix) guide.  Makes it alot easier to create Release Notes - thanks.

**Pull request merged**